### PR TITLE
Be resilient to VS API version changes in navigateto.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @Pilchie

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by Pilchie** [comment]
👍

File: src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToItemProvider.cs [hunk 1]
```diff
@@ -74,5 +71,30 @@ public void StartSearch(INavigateToCallback callback, string searchValue)
...
+        private bool GetSearchCurrentDocumentOption(INavigateToCallback callback)
+        {
+            try
+            {
+                return GetSearchCurrentDocumentOptionWorker(callback);
+            }
+            catch (TypeLoadException)
+            {
+                // The version of the APIs we call in VS may not match what the 
+                // user currently has on the box (as the APIs have changed during
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToItemProvider.cs]: Unmatched parenthesis in the comment. It hurts.

File: src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToItemProvider.cs [hunk 2]
```diff
@@ -74,5 +71,30 @@ public void StartSearch(INavigateToCallback callback, string searchValue)
 
             searcher.Search();
         }
+
+        private bool GetSearchCurrentDocumentOption(INavigateToCallback callback)
+        {
+            try
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToItemProvider.cs]: Do we have a bug tracking the removal of this hack? (Entirely OK with the hack though.)

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi Is this what was breaking the tests over the weekend?

**Pull request merged**